text/0000-spec-requirements.md

```diff
@@ -0,0 +1,53 @@
+# Spec Requirements
+
+## Test writing
+
+### Tests are written as framework-agnostic
+* Should support unit and end-to-end tests
+* JavaScript executed on the DOM is separate from the assertions
+* Tests accept `window` and optional DOM fragment
```

Comment:
Should we generalize the requirement instead of suggesting precisely how the test receives something? The former seems like it will be consistent across version of the specs, while the latter can change depending on implementation.

Reply:
This is an actual requirement. Having the `window` is part of having fully-functional tests which are still exportable.

An example of this would be checkboxes. To test for live updating of `checkbox` after a user (the test runner) clicks the label, we need window because that gives us dynamic updates as opposed to the static `domFragment` which would not know about the update (change in checked state).

Maybe it should be rewritten:

`Individual tests should have access to a component's `window` to allow for testing of some user-initiated actions`